### PR TITLE
[elixir] feat: make schedule dmap task a release task

### PR DESCRIPTION
**Pull request by grejdi-mbta**
This PR moves the ScheduleDmap task from being a Mix task to being a Release task. With the latter we are able to run it as part of AWS Batch or by using the ECS's [run task](https://docs.aws.amazon.com/cli/latest/reference/ecs/run-task.html).

There are 2 options for testing this locally:
Note: If you need to setup your local environment, refer to [README.md](https://github.com/mbta/data_platform/blob/main/README.md).
1. Through tests:
```sh
cd ex_cubic_ingestion
mix test
```
2. Through application:
```sh
cd ex_cubic_ingestion
mix release
_build/dev/rel/ex_cubic_ingestion/bin/ex_cubic_ingestion start
# in new terminal window
_build/dev/rel/ex_cubic_ingestion/bin/ex_cubic_ingestion eval "ReleaseTasks.ScheduleDmap.run(%{})"
# you should see '[info] [ex_cubic_ingestion] [workers] [schedule_dmap]' in the output of the application
```

### CODE REVIEW

File: ex_cubic_ingestion/test/release_tasks/schedule_dmap_test.exs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+defmodule ReleaseTasks.ScheduleDmapTest do
+  use ExCubicIngestion.DataCase, async: true
```
Comment (boringcactus) [inline ex_cubic_ingestion/test/release_tasks/schedule_dmap_test.exs:2]: If I'm remembering right, `async: true` tends to interact poorly with database access, which Oban performs unless [it is set to a testing mode](https://hexdocs.pm/oban/testing.html#setup-application-config), which it looks like [currently isn't happening in the test config](https://github.com/mbta/data_platform/blob/d6615bf9c7dcfcf42a30ab720136550483f482cd/ex_cubic_ingestion/config/test.exs#L9), so for test robustness you should probably add `testing: :inline` to your Oban config in `config/test.exs`.

**Review by boringcactus — APPROVED** [review]
Looks pretty good, just a couple minor questions and suggestions.

File: ex_cubic_ingestion/test/release_tasks/schedule_dmap_test.exs (hunk 1 above)
Comment (grejdi-mbta) [reply ex_cubic_ingestion/test/release_tasks/schedule_dmap_test.exs:2]: I don't actually want to run the job here. Just want to make sure it's queued up. For when I do want to run it, I usually use [perform_job/3](https://hexdocs.pm/oban/Oban.Testing.html#perform_job/3), which will go into `:inline` mode (see [this](https://github.com/sorentwo/oban/blob/v2.12.1/lib/oban/testing.ex#L233). Great feedback!

File: ex_cubic_ingestion/lib/release_tasks/schedule_dmap.ex [hunk 2]
```diff
@@ -0,0 +1,34 @@
+defmodule ReleaseTasks.ScheduleDmap do
+  @moduledoc """
+  This task will insert the ScheduleDmap job upon execution, resulting in
+  the fetching of all DMAP feeds.
+
+  Based on: https://hexdocs.pm/phoenix/releases.html#ecto-migrations-and-custom-commands
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/release_tasks/schedule_dmap.ex:6]: I'm planning on adding more Release tasks and will be keeping this in mind. I'm afraid, the single module might get too big.

File: ex_cubic_ingestion/test/release_tasks/schedule_dmap_test.exs (hunk 1 above)
Comment (boringcactus) [reply ex_cubic_ingestion/test/release_tasks/schedule_dmap_test.exs:2]: Ah, yeah. If memory serves, that's what `testing: :manual` is for.

File: ex_cubic_ingestion/lib/release_tasks/schedule_dmap.ex [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    Application.load(@app)
+
+    # disables running the full app and just start Oban and Ecto
+    Application.put_env(@app, :start_app_children?, false)
+
+    # starts app
+    Application.ensure_all_started(@app)
+
+    # add a new schedule dmap job
+    Oban.insert!(ScheduleDmap.new(args))
```
[the remark this answers is not in the document]
Comment (grejdi-mbta) [reply ex_cubic_ingestion/lib/release_tasks/schedule_dmap.ex:30]: For testing, I wanted to be able to pass something in, so I can make sure I was queueing up the correct job. So figured I could use the args.

[the remark this answers is not in the document]
Comment (boringcactus) [reply ex_cubic_ingestion/lib/release_tasks/schedule_dmap.ex:30]: I'm not sure if that's necessary - I'm skeptical that a failure to enqueue the job in `ReleaseTasks.ScheduleDmap.run` could be masked by something else running at the same time enqueuing a different `ReleaseTasks.ScheduleDmap` job - but I guess there's no harm in having it there.

**Pull request merged**